src/Rules/Variables/UnsetRule.php

```diff
@@ -48,7 +48,7 @@ public function processNode(Node $node, Scope $scope): array
 		return $errors;
 	}
 
-	private function canBeUnset(Node $node, Scope $scope): ?IdentifierRuleError
+	private function canBeUnset(Node $node, Scope $scope, bool $inArrayDimFetch = false): ?IdentifierRuleError
```

Comment:
More logical solution is to exclude `$node instanceof Node\Expr\PropertyFetch` section completely out of `canBeUnset` recursion and just check it directly in processNode.

The other part of this about unsetting on readonly property is also wrong so we need to fix that too:

* https://phpstan.org/r/5c6651e5-cfaf-4688-8b51-0f9ac27c8495
* https://3v4l.org/vONrk#v8.4.4

Reply:
interessting.. 

> Readonly properties cannot be [unset()](https://www.php.net/manual/en/function.unset.php) once they are initialized. However, it is possible to unset a readonly property prior to initialization, from the scope where the property has been declared.

https://www.php.net/manual/en/language.oop5.properties.php